--- /etc/group	Tue Aug 17 20:22:09 2010
+++ group	Fri Aug 20 13:03:53 2010
@@ -1,9 +1,9 @@
-wheel:*:0:root
+wheel:*:0:root,live
 daemon:*:1:daemon
 kmem:*:2:root
 sys:*:3:root
 tty:*:4:root
-operator:*:5:root
+operator:*:5:root,live
 bin:*:7:
 news:*:8:
 wsrc:*:9:
@@ -66,7 +66,7 @@
 _sndio:*:99:
 _ldapd:*:100:
 _iked:*:101:
-dialer:*:117:
+dialer:*:117:live
 nogroup:*:32766:
 nobody:*:32767:
 _iodine:*:598:
